torch/onnx/_internal/fx/passes/fx_to_onnxscript.py

```diff
@@ -178,23 +217,78 @@ def _fill_tensor_meta(
     for i, (onnxscript_value, expected_value) in enumerate(
         zip(flat_onnxscript_values, flat_expected_values)
     ):
-        # Only set shape for now as we don't need type information.
-        onnxscript_value.shape = tuple(expected_value.size())
-        # FIXME(titaiwang): type promotion is not supported yet,
-        # so set dtype will break onnx shape inference.
+        # We set node output sizes to be dynamic to continue the model conversion,
+        # and inputs are also set to be dynamic in add_input().
+        onnxscript_value.shape = tuple(
+            [dim if isinstance(dim, int) else None for dim in expected_value.size()]
+        )
+        # TODO(titaiwang): This could break in terms of lacking of type_promotion now.
         onnxscript_value.dtype = expected_value.dtype
         if i > 0:
             onnxscript_value.name = f"{name}_{i}"
         else:
             onnxscript_value.name = name
 
 
+def _wrap_fx_args_as_torch_args(
+    complete_args: list, complete_kwargs: dict
+) -> Tuple[tuple, dict]:
+    # prepare torch format args and kwargs for op-level validation
+    # Use fake tensor to create real tensor to feed in ops
+    torch_args = []
+    for arg in complete_args:
+        if isinstance(arg, torch.fx.Node):
+            # Create a concreate test tensor based on the fake tensor
+            with torch.utils._mode_utils.no_dispatch():
+                # eg: aten_where needs BOOL in input_args
+                # fx_name_to_onnxscipt_value could help?
+                fake_tensor = arg.meta["val"]
+                if isinstance(fake_tensor, list):
+                    for meta_value in fake_tensor:
+                        # could be varying size
+                        # use 2 to replace dynamic shape
+                        shape = [
+                            dim if isinstance(dim, int) else 2
+                            for dim in meta_value.shape
+                        ]
+                        dtype = (
+                            meta_value.dtype
+                            if meta_value.dtype
+                            not in {torch.int64, torch.int32, torch.uint8, torch.bool}
+                            else torch.float32
+                        )
+                        torch_args.append(torch.randn(meta_value.shape, dtype=dtype))
+                elif isinstance(fake_tensor, torch.Tensor):
+                    # could be varying size
+                    # use 2 to replace dynamic shape
```

Comment:
😱

Reply:
https://github.com/microsoft/onnx-script/issues/393#issuecomment-1478289501

Reply:
Lolll

> 😱

Reply:
Yup, this is a mess. I intend to address this in another PR. Currently, this is under op_level_debug. Use FakeTensorProp on symbolic graph might be the solution!

Reply:
No, `2` is one thing should never appears in the codebase. As suggested in my very early comment, this should be an optional feature and disabled when dynamic shape is enabled. This is reasonable because when there is no real shape, we don't know what inputs to execute on.

Reply:
It's done, please see the options.py